### PR TITLE
GPG-215 Migrate database (changes for PaaS)

### PR DESCRIPTION
**Pull request by jamesgriff**
These are the changes needed to the PaaS instance (to import the data)

### CODE REVIEW

File: GenderPayGap.Database/Models/AuditLog.cs [hunk 1]
```diff
@@ -1,17 +1,31 @@
...
+        public string DetailsString { get; set; }
+
+        [JsonProperty]
+        public long? OrganisationId { get; set; }
+        [JsonProperty]
+        public long? OriginalUserId { get; set; }
+        [JsonProperty]
+        public long? ImpersonatedUserId { get; set; }
+
         public virtual Organisation Organisation { get; set; }
```
Comment (aidayounesi) [inline GenderPayGap.Database/Models/AuditLog.cs:29]: For consistency would it be ok to add [ForeignKey("OrganisationId")]

File: GenderPayGap.WebUI/Controllers/Admin/AdminDataMigrationController.cs [hunk 2]
```diff
@@ -0,0 +1,648 @@
...
+            var fileContentResult = new FileContentResult(allDataBytes, "application/json")
+            {
+                FileDownloadName = "AllData.json.gz"
+            };
+
+            return fileContentResult;
+        }
+
+        [AllowAnonymous]
+        [HttpGet("data-migration/export-all")]
```
Comment (aidayounesi) [inline GenderPayGap.WebUI/Controllers/Admin/AdminDataMigrationController.cs:64]: I'm a bit confused about having two HttpGet with the same url

File: GenderPayGap.WebUI/Controllers/Admin/AdminDataMigrationController.cs [hunk 3]
```diff
@@ -0,0 +1,648 @@
...
+        public AdminDataMigrationController(IDataRepository dataRepository)
+        {
+            this.dataRepository = dataRepository;
+        }
+
+
+        [HttpGet("data-migration/export-all")]
+        public IActionResult ExportAllData()
+        {
+            if (dataRepository.GetAll<User>().Count() > 0)
```
Comment (aidayounesi) [inline GenderPayGap.WebUI/Controllers/Admin/AdminDataMigrationController.cs:40]: Why? 😃

File: GenderPayGap.WebUI/Controllers/Admin/AdminDataMigrationController.cs [hunk 4]
```diff
@@ -0,0 +1,648 @@
...
+        {
+            Response.BodyWriter.WriteAsync(Encoding.UTF8.GetBytes(AdminMigrationTopAndBottomOfView.BottomOfView));
+            Response.CompleteAsync().Wait();
+        }
+
+        private void ImportDataFromJsonString(string fileAsString)
+        {
+            AllData allData = JsonConvert.DeserializeObject<AllData>(fileAsString);
+
+            string deleteAllSql = @"
```
Comment (aidayounesi) [inline GenderPayGap.WebUI/Controllers/Admin/AdminDataMigrationController.cs:283]: I reckon we can delete all records of a table by "DELETE FROM TABLE" without any where clause

File: GenderPayGap.WebUI/Controllers/Admin/AdminDataMigrationController.cs (hunk 2 above)
Comment (jamesgriff) [reply GenderPayGap.WebUI/Controllers/Admin/AdminDataMigrationController.cs:64]: Oops, one of these methods was added as a merge error - I'll remove it.
Thanks for spotting this!

File: GenderPayGap.WebUI/Controllers/Admin/AdminDataMigrationController.cs (hunk 3 above)
Comment (jamesgriff) [reply GenderPayGap.WebUI/Controllers/Admin/AdminDataMigrationController.cs:40]: Similarly, this was an earlier version of the code added as a merge error - removed now.
This logic has now been moved into the better-named `IsUserAllowedToImportData`

File: GenderPayGap.WebUI/Controllers/Admin/AdminDataMigrationController.cs (hunk 4 above)
Comment (jamesgriff) [reply GenderPayGap.WebUI/Controllers/Admin/AdminDataMigrationController.cs:283]: I tried to do `DELETE FROM Table` in the UI (DataGrip) but got an error `Unsafe query: 'Delete' statement without 'where' clears all data in the table`.
Maybe this is just in the UI though?

File: GenderPayGap.Database/Models/AuditLog.cs (hunk 1 above)
Comment (jamesgriff) [reply GenderPayGap.Database/Models/AuditLog.cs]: Ah yes, good idea

**Comment by jamesgriff** [comment]
> Question: are we going to remove this code after migration?
I expect we'll remove this code after migration.
Unless we find it very useful for something else (although I'd then want to re-visit it to re-think it's production-worthiness!)

File: GenderPayGap.WebUI/Controllers/Admin/AdminDataMigrationController.cs (hunk 4 above)
Comment (aidayounesi) [reply GenderPayGap.WebUI/Controllers/Admin/AdminDataMigrationController.cs:271]: I'm fairly sure it's just the UI

**Pull request merged**